### PR TITLE
feat(sdk): Improve JS SDK configuration

### PR DESCRIPTION
**Pull request by dcramer**
- Reuse existing environment / release configuration
- Add SENTRY_FRONTEND_WHITELIST_URLS setting

### CODE REVIEW

**Comment by dcramer** [comment]
This is to prep for my feedback around https://github.com/getsentry/getsentry/pull/3187

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -915,6 +915,9 @@ def create_partitioned_queues(name):
 # over SENTRY_FRONTEND_PROJECT or SENTRY_PROJECT
 SENTRY_FRONTEND_DSN = None
 
+# Configuration for JavaScript's whitelistUrls - defaults to ALLOWED_HOSTS
+SENTRY_FRONTEND_WHITELIST_URLS = None
```
Comment (dcramer) [inline src/sentry/conf/server.py]: Open to other ideas here, but i dont think ALLOWED_HOSTS can be relied on for our production usage (as we bind some internal domains which AFAICT wouldnt make sense). Might be worth chatting live/Slack tho to see if that is true..

Current sentry prod whitelist:

```
whitelistUrls: ['sentry.io', 's1.sentry-cdn.com', 'dev.getsentry.net', '<anonymous>'],
```

File: src/sentry/web/client_config.py [hunk 2]
```diff
@@ -160,10 +160,13 @@ def get_client_config(request=None):
...
         "sentryConfig": {
             "dsn": public_dsn,
-            "release": version_info["build"],
+            "release": settings.SENTRY_SDK_CONFIG["release"],
+            "environment": settings.SENTRY_SDK_CONFIG["environment"],
             # By default `ALLOWED_HOSTS` is [*], however the JS SDK does not support globbing
-            "whitelistUrls": list(
-                "" if settings.ALLOWED_HOSTS == ["*"] else settings.ALLOWED_HOSTS
+            "whitelistUrls": (
+                settings.SENTRY_FRONTEND_WHITELIST_URLS
```
Comment (dcramer) [inline src/sentry/web/client_config.py]: could make this check for None if we really care, but doesnt actually matter in our production cases

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Pull request merged**